cognito.tf

```diff
@@ -0,0 +1,187 @@
+# Cognito resources used by the Elasticsearch domain
+
+# Cognito users
+resource "aws_cognito_user" "dmarc" {
+  for_each = { for k, v in var.cognito_usernames : k => v }
+
+  user_pool_id = aws_cognito_user_pool.dmarc.id
+  username     = each.key
+  attributes = {
+    email = each.value["email"]
+  }
+}
+
+# The Cognito user pool
+resource "aws_cognito_user_pool" "dmarc" {
+  account_recovery_setting {
+    recovery_mechanism {
+      name     = "verified_email"
+      priority = 1
+    }
+  }
+
+  admin_create_user_config {
+    allow_admin_create_user_only = true
+  }
+
+  auto_verified_attributes = [
+    "email",
+  ]
+
+  mfa_configuration = "ON"
+  name              = var.cognito_user_pool_name
+  software_token_mfa_configuration {
+    enabled = true
+  }
+  user_attribute_update_settings {
+    attributes_require_verification_before_update = [
+      "email",
+    ]
+  }
+}
+
+# The managed Cognito user pool client for the Elasticsearch endpoint. This
+# resource is used to manage the Cognito user pool client that is automatically
+# created by Elasticsearch (OpenSearch) when Cognito authentication is enabled.
+# Thanks to this comment for pointing me in the right direction:
+# https://github.com/hashicorp/terraform-provider-aws/issues/5557#issuecomment-1015731466
+resource "aws_cognito_managed_user_pool_client" "dmarc" {
```

Comment:
When I was looking at https://github.com/cisagov/dmarc-import-tf-module/pull/49/files#r1442602252 again I looked at the [`aws_cognito_managed_user_pool_client` documentation](https://registry.terraform.io/providers/hashicorp/aws/4.67.0/docs/resources/cognito_user_pool_client) and saw this blurb:
> To manage a User Pool Client created by another service, such as when [configuring an OpenSearch Domain to use Cognito authentication](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/cognito-auth.html), use the [aws_cognito_managed_user_pool_client resource](https://registry.terraform.io/providers/hashicorp/aws/4.67.0/docs/resources/cognito_managed_user_pool_client) instead.


Since the examples sounds exactly like what we're doing Is there a reason we're not using that resource instead?

Reply:
I am confused.  I am using the `aws_cognito_managed_user_pool_client` resource here for exactly that reason.  Am I missing something here or are you?

Reply:
Nope, I'm just having a Monday moment and landed on the wrong documentation page x.x